### PR TITLE
Use filters to clean up Liquid template

### PR DESCRIPTION
**Pull request by pathawks**
Obviously, we won't be able to ship this until Jekyll 3.3 lands, but I was so excited about the new filters I wanted to see how we could use them to clean up our template.

We will need to drop support for anything earlier than Jekyll 3.3, but I feel the advantages are worth it. Perhaps we wait until GitHub Pages supports Jekyll 3.3?

jekyll/jekyll#5399

### CODE REVIEW

File: jekyll-sitemap.gemspec [hunk 1]
```diff
@@ -16,7 +16,8 @@ Gem::Specification.new do |spec|
 
   spec.add_runtime_dependency "addressable", "~>2.4.0"
 
-  spec.add_development_dependency "jekyll", ">= 2.0"
+  spec.add_dependency "jekyll", ">= 3.3.0"
```
Comment (benbalter) [inline jekyll-sitemap.gemspec]: Could this be `~> 3.3`?

File: lib/sitemap.xml [hunk 2]
```diff
@@ -1,19 +1,14 @@
...
   {% for post in site.posts %}{% unless post.sitemap == false %}
   <url>
-    <loc>{{ post.url | prepend: site_url | normalize_url }}</loc>
-    {% if post.last_modified_at %}
-    <lastmod>{{ post.last_modified_at | date_to_xmlschema }}</lastmod>
-    {% else %}
-    <lastmod>{{ post.date | date_to_xmlschema }}</lastmod>
-    {% endif %}
+    <loc>{{ post.url | absolute_url }}</loc>
+    <lastmod>{{ post.last_modified_at | default: post.date | date_to_xmlschema }}</lastmod>
```
Comment (benbalter) [inline lib/sitemap.xml]: Should this still be wrapped in a conditional?

Comment (pathawks) [inline lib/sitemap.xml]: Any `post` is going to have a date. The conditional was only to see if it had a `last_modified_at` and would use that instead if available. This is exactly what the `default` filter does, and lets us avoid the conditional.

**Comment by pathawks** [comment]
> As a reminder, what was the goal behind `normalize_url`?

Added in #107 to make sure that URLs were escaped once and only once. The new URL filters [take care of this for us](https://github.com/jekyll/jekyll/blob/v3.3.0/lib/jekyll/filters/url_filters.rb#L15) and I am so thrilled about that!

**Comment by pathawks** [comment]
@benbalter Rebased and addressed your comments. Could you just look over one more time please.

**Comment by benbalter** [comment]
@jekyllbot: merge +minor

Thanks @pathawks! Another awesome PR.

**Pull request merged**